source/geometry_model/spherical_shell.cc

```diff
@@ -45,22 +46,170 @@ namespace aspect
+              // one slice at a time. The z-axis value are defined in slices_coordinates
+              for (unsigned int point_layer = 0; point_layer < R_values.size(); ++point_layer)
+                {
+                  for (unsigned int vertex_n = 0; vertex_n < sphere_mesh.n_vertices();
+                       ++vertex_n)
+                    {
+                      const Point<dim> vertex = sphere_mesh.get_vertices()[vertex_n];
+                      points[point_layer * sphere_mesh.n_vertices() + vertex_n] =
+                        vertex * R_values[point_layer];
+                    }
+                }
+
+              // Then create the cells of each of the slices, one stack at a
+              // time
+              std::vector<CellData<dim>> cells;
+              cells.reserve((R_values.size() - 1) * sphere_mesh.n_active_cells());
+
+              SubCellData               subcell_data;
+
+              for (const auto &cell : sphere_mesh.active_cell_iterators())
+                {
+                  for (unsigned int cell_layer = 0; cell_layer < R_values.size() - 1; ++cell_layer)
+                    {
+                      CellData<dim> this_cell;
+                      for (unsigned int vertex_n = 0;
+                           vertex_n < GeometryInfo<dim-1>::vertices_per_cell;
+                           ++vertex_n)
+                        {
+                          this_cell.vertices[vertex_n] =
+                            cell->vertex_index(vertex_n) + cell_layer * sphere_mesh.n_vertices();
+                          this_cell.vertices[vertex_n + GeometryInfo<dim-1>::vertices_per_cell] =
+                            cell->vertex_index(vertex_n) +
+                            (cell_layer + 1) * sphere_mesh.n_vertices();
+                        }
+
+                      this_cell.material_id = cell->material_id();
+
+#if !DEAL_II_VERSION_GTE(9,1,0)
+                      // In deal.II prior to version 9.1, the hyper_ball mesh generated above
+                      // had cells that were inconsistently oriented: Some cells had a normal
+                      // vector that pointed to the inside of the ball, some that pointed
+                      // to the outside. This leads to cells with either negative or
+                      // positive volume if we extrude them in the radial direction. We need
+                      // to fix this up.
+
+                      if (GridTools::cell_measure (points, this_cell.vertices) < 0)
+                        {
+                          if (dim == 2)
+                            std::swap (this_cell.vertices[1], this_cell.vertices[2]);
+                          if (dim == 3)
+                            {
+                              std::swap (this_cell.vertices[1], this_cell.vertices[2]);
+                              std::swap (this_cell.vertices[5], this_cell.vertices[6]);
+                            }
+                        }
+#endif
+
```

Comment:
Can you add ``Assert(GridTools::cell_measure (points, this_cell.vertices) < 0, ExcInternalError());`` here please?